src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -365,7 +365,7 @@ def get_raw_token(self, resource=None, scopes=None, subscription=None, tenant=No
         if identity_type:
             # managed identity
             if tenant:
-                raise CLIError("Tenant shouldn't be specified for managed identity account")
+                tenant = False
```

Comment:
See https://github.com/Azure/azure-cli/issues/26775#issuecomment-2311983442

Silently discarding `tenant` for managed identity may lead to unexpected and misleading behaviors.

Reply:
Great feedback, I don't have time to look into strategic solution for this but the fundamental functionality of managed identities for artifacts is broken because of this so I'm not sure who/if there's someone to look at it ?